paymentsheet/src/main/java/com/stripe/android/paymentsheet/repositories/ElementsSessionRepository.kt

```diff
@@ -155,6 +153,40 @@ private fun PaymentSheet.CustomerConfiguration.toElementSessionParam(): String?
     }
 }
 
+private fun ElementsSessionParams.DeferredIntentType.toStripeIntent(options: ApiRequest.Options): StripeIntent {
+    val deferredIntentParams = this.deferredIntentParams
+    val now = Calendar.getInstance().timeInMillis
+    return when (deferredIntentParams.mode) {
+        is DeferredIntentParams.Mode.Payment -> PaymentIntent(
+            id = deferredIntentParams.paymentMethodConfigurationId,
+            paymentMethodTypes = deferredIntentParams.paymentMethodTypes,
+            amount = (deferredIntentParams.mode as DeferredIntentParams.Mode.Payment).amount,
```

Comment:
I think you can assign this in the when to avoid the need to cast here.

Reply:
thanks! Updated